### PR TITLE
*: set length and decimal for constant values

### PR DESCRIPTION
**Pull request by XuHuaiyu**
as title decribed
PTAL @hanfei1991 @shenli

### CODE REVIEW

File: mysql/const.go [hunk 1]
```diff
@@ -195,6 +195,10 @@ const AllPrivMask = AllPriv - 1
 
 // MySQL type maximum length.
 const (
+	// For arguments that have no fixed number of decimals, the decimals value is set to 31,
```
Comment (shenli) [inline mysql/const.go]: Is this the same behaviour with MySQL?

File: util/types/field_type.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ import (
 
 	"github.com/pingcap/tidb/mysql"
 	"github.com/pingcap/tidb/util/charset"
+	"strconv"
```
Comment (shenli) [inline util/types/field_type.go]: Move this up.

File: util/types/field_type.go [hunk 3]
```diff
@@ -178,46 +179,100 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
 	switch x := value.(type) {
 	case nil:
 		tp.Tp = mysql.TypeNull
```
Comment (hanfei1991) [inline util/types/field_type.go]: mysql> create table t_null as select null;
Query OK, 1 row affected (0.08 sec)
Records: 1  Duplicates: 0  Warnings: 0

mysql> desc t_null;
+-------+-----------+------+-----+---------+-------+
| Field | Type      | Null | Key | Default | Extra |
+-------+-----------+------+-----+---------+-------+
| NULL  | binary(0) | YES  |     | NULL    |       |
+-------+-----------+------+-----+---------+-------+
1 row in set (0.00 sec)

File: mysql/const.go (hunk 1 above)
Comment (XuHuaiyu) [inline mysql/const.go]: yes

File: util/types/field_type.go (hunk 3 above)
Comment (XuHuaiyu) [inline util/types/field_type.go]: yeah? I don't really understand what you mean

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**